src/vdom/diff.js

```diff
@@ -216,21 +229,23 @@ function innerDiffNode(dom, vchildren, context, mountAll, isHydrating) {
 				}
 			}
 
+			if (vchild.nodeName === Fragment) {
+				innerDiffNode(dom, vchild.children, context, mountAll, isHydrating);
+				return;
+			}
+
 			// morph the matched/found/created DOM child to match vchild (deep)
 			child = idiff(child, vchild, context, mountAll);
 
 			f = originalChildren[i];
-			if (child && child!==dom && child!==f) {
-				if (f==null) {
-					dom.appendChild(child);
-				}
-				else if (child===f.nextSibling) {
-					removeNode(f);
-				}
-				else {
-					dom.insertBefore(child, f);
+			if (child instanceof Array) {
+				for (let i = 0; i < child.length; i++) {
+					updateChild(dom, child[i], f);
 				}
 			}
+			else {
+				updateChild(dom, child, f);
```

Comment:
Likely a small performance loss, because `updateChild` is not inlined anymore. Need to measure this.

Reply:
Compared with `master` my measurements don't show much differences. The `empty diff` test shows the most difference.

| | master | branch | 
|---|---|---|
| empty diff | 89503/s (30 ticks) | 84007/s (31 ticks) |
| repeat diff | 11707/s (228 ticks) | 11373/s (228 ticks) |
| large VTree | 43276/s (62 ticks) | 43877/s (58 ticks) |
| style/prop mutation | 19974/s (133 ticks) | 19745/s (132 ticks) |
| SSR Hydrate | 1384/s (1962 ticks) | 1364/s (1903 ticks) |

Reply:
Fixed the slowdown of the `empty diff` check. Problem was the `Array.isArray` check at https://github.com/developit/preact/pull/1080/files#diff-a0896fa5d65b1d80ca4d496badbae50fR106

Reply:
how did you run those tests?